src/Crypto/WebAuthn/Model/Kinds.hs

```diff
@@ -69,19 +68,23 @@ data CeremonyKind
     -- for an introductory overview and
     -- [§ 1.3.3 Authentication](https://www.w3.org/TR/webauthn-2/#sctn-sample-authentication) for implementation examples.
     Authentication
+  deriving (Eq, Show)
 
 -- | The singleton type for 'CeremonyKind'
 data SCeremonyKind :: CeremonyKind -> Type where
   SRegistration :: SCeremonyKind 'Registration
   SAuthentication :: SCeremonyKind 'Authentication
 
-deriving instance Show (SCeremonyKind c)
+instance SingKind CeremonyKind where
+  type Demote CeremonyKind = CeremonyKind
+  fromSing SRegistration = Registration
+  fromSing SAuthentication = Authentication
+  toSing Registration = SomeSing SRegistration
+  toSing Authentication = SomeSing SAuthentication
```

Comment:
I think it's a bit overkill to make use of this additional singletons functionality when we can't make good use of it. There's only two uses of `demote`, and we can't replace any other `sing`'s with `demote`. So I'd suggest not using `SingKind` and `demote`.

In addition, I think it's neat that the singleton types are only used as types, while the corresponding kinds are only used as kinds. Because of this, the kinds also don't need to have an `Eq` or `Show` instance, because they are never used at runtime. The singleton types however are used at runtime, which is why they should have an `Eq` and `Show` instance.

Reply:
Sounds good. My intuition is that `S*` singleton types are more appropriate when you need dependently typed expressions, and demotion is more appropriate when you just need a value. I haven't used singletons much, though. Do you know of a guide to writing idiomatic code with singletons?

Reply:
The best introduction to singletons I read was [this one](https://blog.jle.im/entry/introduction-to-singletons-1.html). Personally for this project I only bothered to know one thing about singletons: If you want to know what a type is at runtime, call `sing @<type>` on it and pattern match on the result.